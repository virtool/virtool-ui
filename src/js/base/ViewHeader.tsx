import React from "react";
import { Helmet, HelmetProvider } from "react-helmet-async";
import styled from "styled-components";

const StyledViewHeader = styled.header`
    display: block;
    margin: 10px 0 20px;
`;

StyledViewHeader.displayName = "StyledViewHeader";

type ViewHeaderProps = {
    children?: React.ReactNode;
    className?: string;
    title: string;
};

export function ViewHeader({ className, title, children }: ViewHeaderProps) {
    return (
        <StyledViewHeader className={className}>
            <HelmetProvider>
                <Helmet>
                    <title>{title}</title>
                </Helmet>
                {children}
            </HelmetProvider>
        </StyledViewHeader>
    );
}
