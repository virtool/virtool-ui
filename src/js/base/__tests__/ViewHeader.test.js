import { ViewHeader } from "../ViewHeader";

describe("<ViewHeader />", () => {
    let wrapper;

    it("should render with title", () => {
        wrapper = shallow(<ViewHeader title="Foo" />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render with count", () => {
        wrapper = shallow(<ViewHeader title="Foo" count={10} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render renders children", () => {
        wrapper = shallow(
            <ViewHeader title="test-child">
                <div>Hello world</div>
            </ViewHeader>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
