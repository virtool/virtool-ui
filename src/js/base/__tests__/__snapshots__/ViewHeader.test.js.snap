// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewHeader /> should render renders children 1`] = `
<ViewHeader__StyledViewHeader>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      test-child
    </title>
  </HelmetWrapper>
  <div>
    Hello world
  </div>
</ViewHeader__StyledViewHeader>
`;

exports[`<ViewHeader /> should render with count 1`] = `
<ViewHeader__StyledViewHeader>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Foo
    </title>
  </HelmetWrapper>
</ViewHeader__StyledViewHeader>
`;

exports[`<ViewHeader /> should render with title 1`] = `
<ViewHeader__StyledViewHeader>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Foo
    </title>
  </HelmetWrapper>
</ViewHeader__StyledViewHeader>
`;
